File: cs12b/pa3/DictionaryTest.java
//------------------------------------------------------------------------------------------------
// DictionaryTest.java
// Aryan Samuel
// arsamuel
// CMPS 12B
// 04/24/16
// Programming Assignment 3
// A test client for the Dctionary ADT. Each new Test block shows what was added only
// after the previous test block passed i.e. no erros when compiled, expected output.
//------------------------------------------------------------------------------------------------

public class DictionaryTest {
	public static void main(String[] args) {
		Dictionary A = new Dictionary();
		System.out.println(A.isEmpty());
		// Test 1
		A.insert("1","a");
		// Test 2
		A.insert("2","b");
		A.insert("3","c");
		A.insert("4","d");
		A.insert("5","e");
		
		System.out.println(A);
		// Test 3
		System.out.println(A.size());
		System.out.println(A.isEmpty());
		System.out.println();
		
		// Test 4 / DuplicateKeyException Test
		//A.insert("1","a");
		
		// Test 5
		System.out.println(A.lookup("1"));
		System.out.println(A.lookup("4"));
		System.out.println();
		
		// Test 6
		A.delete("3");
		System.out.println(A);
		System.out.println(A.size());
		System.out.println();
		
		// Test 7 / KeyNotFoundException Test
		//A.delete("6");
		
		// Test 8
		A.makeEmpty();
		System.out.println(A.size());
		System.out.println(A);
	}
}

File: DictADT/DictionaryTest.c
//--------------------------------------------------------------------------------------------------
// DictionaryTest.c
// Aryan Samuel
// arsamuel
// CMPS 12B
// 05/21/16
// Programming Assignment 5
// A test client for the Dctionary ADT. Each new Test block shows what was added only
// after the previous test block passed i.e. no erros when compiled, expected output.
//--------------------------------------------------------------------------------------------------

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char* argv[]) {
	Dictionary A = newDictionary();

	// Test 1
	insert(A,"1","one");
	// Test 2
	insert(A,"2", "two");
	insert(A,"3", "three");
	insert(A,"4", "four");
	insert(A,"5", "five");
	
	printDictionary(stdout,A);
	
	// Test 3
	printf("%d\n",size(A));
	printf("%d\n",isEmpty(A));
	printf("\n");
	
	// Test 4
	printf("%s\n", lookup(A,"1"));
	printf("%s\n", lookup(A,"3"));
	printf("\n");
	
	// Test 5 / Duplicate Test
	//insert(A,"2","two");
	
	// Test 6
	delete(A,"4");
	printDictionary(stdout,A);
	printf("%d\n",size(A));
	printf("\n");
	
	// Test 7 / Non-existent key test
	//delete(A,"7");
	
	// Test 8
	makeEmpty(A);
	printf("%d\n",isEmpty(A));
	//printDictionary(stdout,A); 
	
	freeDictionary(&A);
	return (EXIT_SUCCESS);
}